rwengine/CMakeLists.txt

```diff
@@ -1,12 +1,66 @@
-FILE(GLOB_RECURSE RENDERWARE_SOURCES "src/*.cpp")
-FILE(GLOB_RECURSE RENDERWARE_HEADERS "include/*.hpp")
-
-source_group("Header Files" FILES ${RENDERWARE_HEADERS})
-source_group("Source Files" FILES ${RENDERWARE_SOURCES})
+set(RWENGINE_SOURCES
+	src/ai/AIGraph.cpp
+	src/ai/AIGraphNode.cpp
+	src/ai/CharacterController.cpp
+	src/ai/DefaultAIController.cpp
+	src/ai/PlayerController.cpp
+	src/ai/TrafficDirector.cpp
+	src/audio/alCheck.cpp
+	src/audio/MADStream.cpp
+	src/audio/SoundManager.cpp
+	src/BinaryStream.cpp
+	src/core/Logger.cpp
+	src/core/Profiler.cpp
+	src/data/Chase.cpp
+	src/data/ObjectData.cpp
+	src/data/Skeleton.cpp
+	src/dynamics/CollisionInstance.cpp
+	src/engine/Animator.cpp
+	src/engine/GameData.cpp
+	src/engine/GameState.cpp
+	src/engine/GameWorld.cpp
+	src/engine/SaveGame.cpp
+	src/engine/ScreenText.cpp
+	src/items/WeaponItem.cpp
+	src/loaders/DataLoader.cpp
+	src/loaders/GenericDATLoader.cpp
+	src/loaders/LoaderCOL.cpp
+	src/loaders/LoaderCutsceneDAT.cpp
+	src/loaders/LoaderGXT.cpp
+	src/loaders/LoaderIDE.cpp
+	src/loaders/LoaderIFP.cpp
+	src/loaders/LoaderIPL.cpp
+	src/loaders/WeatherLoader.cpp
+	src/objects/CharacterObject.cpp
+	src/objects/CutsceneObject.cpp
+	src/objects/GameObject.cpp
+	src/objects/InstanceObject.cpp
+	src/objects/ItemPickup.cpp
+	src/objects/PickupObject.cpp
+	src/objects/ProjectileObject.cpp
+	src/objects/VehicleObject.cpp
+	src/render/DebugDraw.cpp
+	src/render/GameRenderer.cpp
+	src/render/GameShaders.cpp
+	src/render/MapRenderer.cpp
+	src/render/ObjectRenderer.cpp
+	src/render/OpenGLRenderer.cpp
+	src/render/TextRenderer.cpp
+	src/render/VisualFX.cpp
+	src/render/WaterRenderer.cpp
+	src/script/modules/GameModule.cpp
+	src/script/modules/ObjectModule.cpp
+	src/script/modules/VMModule.cpp
+	src/script/SCMFile.cpp
+	src/script/ScriptDisassembly.cpp
+	src/script/ScriptMachine.cpp
+	src/script/ScriptModule.cpp
+	src/script/ScriptTypes.cpp
+	src/TextureArchive.cpp
+)
```

Comment:
does this drop the headers? why?

Also this should probably be a seperate branch imho. A PR could probably be fast-tracked + build/makefile changes often cause regressions for some people, so we should attempt to isolate those.